### PR TITLE
[UnityAds] Adapter updates for 3.7.1 release

### PR DESCRIPTION
**Pull request by richardh-unity3d**

### CODE REVIEW

File: UnityAds/UnityAdsInterstitialCustomEvent.m [hunk 1]
```diff
@@ -28,33 +28,48 @@ @implementation UnityAdsInterstitialCustomEvent
 @dynamic localExtras;
 @dynamic hasAdAvailable;
 
+- (NSString *) getAdNetworkId {
+    return (self.placementId != nil) ? self.placementId : @"";
+}
+
+- (NSError *)createErrorWith:(NSString *)description andReason:(NSString *)reaason andSuggestion:(NSString *)suggestion {
```
Comment (p158276) [inline UnityAds/UnityAdsInterstitialCustomEvent.m:35]: typo for `reason`

File: UnityAds/UnityAdsInterstitialCustomEvent.m [hunk 2]
```diff
@@ -28,33 +28,48 @@ @implementation UnityAdsInterstitialCustomEvent
...
-- (BOOL)hasAdAvailable {
-    return [UnityAds isReady:self.placementId];
+- (BOOL)isRewardExpected {
+    return NO;
 }
 
-- (NSString *) getAdNetworkId {
-    return (self.placementId != nil) ? self.placementId : @"";
+- (BOOL)hasAdAvailable {
+    return _placementId != nil;
```
Comment (p158276) [inline UnityAds/UnityAdsInterstitialCustomEvent.m:57]: Is the `isReady` being depreciated in 3.7.1? How to tell when the ad is expired?

File: UnityAds/UnityAdsRewardedVideoCustomEvent.m [hunk 3]
```diff
@@ -40,24 +40,30 @@ - (void)initializeSdkWithParameters:(NSDictionary *)parameters {
...
+{
+    return NO;
+}
+
 - (BOOL)isRewardExpected
 {
     return YES;
 }
 
 - (BOOL)hasAdAvailable {
```
Comment (p158276) [inline UnityAds/UnityAdsRewardedVideoCustomEvent.m:59]: Is the `isReady` being depreciated in 3.7.1? How to tell when the ad is expired?

Comment (alex-crowe) [reply UnityAds/UnityAdsRewardedVideoCustomEvent.m:59]: Basically you can call load, and if the ad unit is ready we will return response immediately.
or show call will return you the error.

File: UnityAds/UnityAdsInterstitialCustomEvent.m (hunk 2 above)
Comment (richardh-unity3d) [reply UnityAds/UnityAdsInterstitialCustomEvent.m:57]: The desired flow will be to attempt to show the ad if the placement loaded successfully, and if the placement is no longer available by the time show is called, the `unityAdsShowFailed` handler will be called with the error `kUnityShowErrorNotReady`.

File: UnityAds/UnityAdsRewardedVideoCustomEvent.m (hunk 3 above)
Comment (richardh-unity3d) [reply UnityAds/UnityAdsRewardedVideoCustomEvent.m:59]: I would just add that you will know the ad unit is ready after calling load and `unityAdsAdLoaded` is called.  After that the ad unit can be shown and if the ad unit is expired `unityAdsShowFailed` will be called with the error kUnityShowErrorNotReady.

Comment (p158276) [reply UnityAds/UnityAdsRewardedVideoCustomEvent.m:59]: Thanks for the clarification.

**Comment by p158276** [comment]
Hi @richardh-unity3d , the change have been implemented in https://github.com/mopub/mopub-ios-mediation/commit/3aabdf076a72159b68361d46725153f0e7e82cb2
Close out the PR here.

**Pull request closed without merging**